src/Symfony/Component/Console/Application.php

```diff
@@ -81,13 +81,16 @@ class Application implements ResetInterface
     private $initialized;
     private $signalRegistry;
     private $signalsToDispatchEvent = [];
+    private $dontRunPutenv = false;
+    
 
-    public function __construct(string $name = 'UNKNOWN', string $version = 'UNKNOWN')
+    public function __construct(string $name = 'UNKNOWN', string $version = 'UNKNOWN', bool $dontRunPutenv = false)
```

Comment:
I would not add an argument there as this is an edge case. Can't we automatically detect that it is disable?

Reply:
it is possible to check if a function is disabled programmatically, but it doesn't look that pretty: `$disabled = in_array('putenv', array_map('trim', explode(',', ini_get('disable_functions'))))`.

Reply:
using `function_exists()` + `@` in front of the call is enough to work around a disabled `putenv()` (at least that's what we do in other similar cases).

Reply:
using `function_exists()` will always return true even if the function is disabled, so maybe `@` is enough?